policies/lib/tree/30_generic_methods/variable_string_from_command.cf

```diff
@@ -40,48 +40,34 @@
 bundle agent variable_string_from_command(prefix, name, command)
 {
   vars:
-      "old_class_prefix"  string => canonify("variable_string_from_command_${name}");
-      "args"               slist => { "${prefix}", "${name}", "${command}" };
-      "report_param"      string => join("_", args);
-      "full_class_prefix" string => canonify("variable_string_from_command_${report_param}");
-      "class_prefix"      string => string_head("${full_class_prefix}", "1000");
-
-    # means < 8.0, promises.cf breaks on >= 3.14
-    # can be removed when 7.X compat is dropped
-    cfengine_3_15|cfengine_3_18::
-      # only capture stdout for compatibility with previous implementation
-      # add a comment to allow executing the exact same command without hitting
-      # https://northerntech.atlassian.net/browse/CFE-4244
-      "result" data => execresult_as_data("${command} # as_data", "useshell", "stdout");
-    !(cfengine_3_15|cfengine_3_18)::
-      "result" data => execresult_as_data("${command}", "useshell", "stdout");
+      "class_prefix" string => canonify("variable_string_from_command_${name}");
+      "result"         data => execresult_as_data("${command}", "useshell", "stdout");
 
     returned_zero::
       "${prefix}.${name}" string => "${result[output]}";
 
   classes:
     pass1::
-      "returned_zero"     expression => strcmp("${result[exit_code]}", "0");
+      "returned_zero"    expression => strcmp("${result[exit_code]}", "0");
     pass2::
-      "variable_defined"  expression => isvariable("${prefix}.${name}");
+      "variable_defined" expression => isvariable("${prefix}.${name}");
```

Comment:
I'm almost sure those classes should be suffixed by report_data.reportid to ensure they are defined for our correct iteration

Reply:
For some reasons it breaks the method if I add the report in the classes